cdm/src/main/java/ucar/nc2/ft2/coverage/simpgeometry/SimpleGeometryCoverage.java

```diff
@@ -0,0 +1,228 @@
+import ucar.nc2.ft2.coverage.simpgeometry.CFGEOMETRY;
+import ucar.nc2.ft2.coverage.CoverageReader;
+import ucar.nc2.ft2.coverage.SubsetParams;
+import ucar.nc2.ft2.coverage.adapter.SimpleGeometryCS;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Formatter;
+import java.util.List;
+
+/**
+ * SimpleGeometry - forked from Coverage.java
+ * Immutable after setCoordSys() is called.
+ *
+ * @author Katie
+ * @author Carron
+ * @since 8/13/2018
+ */
+// @Immutable
+public class SimpleGeometryCoverage implements VariableSimpleIF, IsMissingEvaluator { //add interface with geometry info? //subsetting data will be diffferent?? is it subset now
+  private final String name;
+  private final DataType dataType;
+  private final AttributeContainerHelper atts;
+  private final String units, description;
+  private final String coordSysName;
+  protected final CoverageReader reader;
+  protected final Object user;
+  
+  private final CFGEOMETRY geometry; //use enum?
+
+  private SimpleGeometryCS coordSys; // almost immutable use coordsys that winor made?
+
+  public SimpleGeometryCoverage(String name, DataType dataType, List<Attribute> atts, String coordSysName, String units, String description, CoverageReader reader, Object user, CFGEOMETRY geometry) {
+    this.name = name;
+    this.dataType = dataType;
+    this.atts = new AttributeContainerHelper(name, atts);
+    this.coordSysName = coordSysName;
+    this.units = units;
+    this.description = description;
+    this.reader = reader;
+    this.user = user;
+    this.geometry = geometry;
+  }
+
+
+  void setCoordSys (SimpleGeometryCS coordSys) {
+    if (this.coordSys != null) throw new RuntimeException("Cant change coordSys once set");
+    this.coordSys = coordSys;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+
+  @Override
+  public DataType getDataType() {
+    return dataType;
+  }
+
+  @Override
+  public List<Attribute> getAttributes() {
+    return atts.getAttributes();
+  }
+
+  @Override
+  public Attribute findAttributeIgnoreCase(String name) {
+    return atts.findAttributeIgnoreCase(name);
+  }
+
+  @Override
+  public String getUnitsString() {
+    return units;
+  }
+
+  @Override
+  public String getDescription() {
+    return description;
+  }
+
+  public String getCoordSysName() {
+    return coordSysName;
+  }
+
+  public Object getUserObject() {
+    return user;
+  }
+  public CFGEOMETRY getGeometry() {
+	  return geometry; 
+  }
+  public String getGeometryDescription() {
+	  switch (geometry) {
+	  
+	  case CFPOINT:
+		  return "Point";
+	  case CFLINE:
+		  return "Line";
+	  case CFPOLYGON:
+		  return "Polygon";
+	  default:
+		  return "";
+	  }
+  }
+  @Override
+  public String toString() {
+    Formatter f = new Formatter();
+    Indent indent = new Indent(2);
+    toString(f, indent);
+    return f.toString();
+  }
+
+  public void toString(Formatter f, Indent indent) {

```

Comment:
I haven't seen Formatter used in toString methods like this before. It's kinda cool!